packages/lesswrong/components/ea-forum/EAHomeMainContent.tsx

```diff
@@ -162,59 +50,9 @@ const EAHomeMainContent = ({FrontpageNode, classes}:{
   FrontpageNode: ComponentType,
   classes: ClassesType
 }) => {
-  // we use the widths of the tabs window and the underlying topics bar
-  // when calculating how far to scroll left and right
-  const tabsWindowRef = useRef<HTMLDivElement|null>(null)
-  const topicsBarRef = useRef<HTMLDivElement|null>(null)
-
-  // we store the topic bar scrollLeft offsets that correspond to displaying each "set" of topics
-  const offsets = useRef<Array<number>>([0])
-
-  useEffect(() => {
-    if (!tabsWindowRef.current || !topicsBarRef.current) return
-    offsets.current = [0]
-    Array.from(topicsBarRef.current.children).forEach((topic: HTMLElement) => {
-      if (!tabsWindowRef.current || !topicsBarRef.current) return
-      // we are looking for the topic that would get cut off at the end of each "set",
-      // by checking if the right edge would be past the window
-      // - if so, this will be the first in the next "set"
-      if (topic.offsetLeft + topic.offsetWidth - offsets.current[offsets.current.length-1] > tabsWindowRef.current.offsetWidth) {
-        // subtract 30px to account for the fade on the left side of the tabs window
-        offsets.current.push(topic.offsetLeft - 30)
-      }
-    })
-  }, [tabsWindowRef, topicsBarRef])
-  
-  const { results: coreTopics } = useMulti({
-    terms: {view: "coreTags"},
-    collectionName: "Tags",
-    fragmentName: 'TagDetailsFragment',
-    limit: 40
-  })
-  let allTabs: TopicsBarTab[] = useMemo(() => {
-    let tabs: TopicsBarTab[] = [frontpageTab]
-    if (coreTopics) {
-      // list the core topics based on topicTabsOrder
-      topicTabsOrder.forEach(topicId => {
-        const topic = coreTopics.find(t => t._id === topicId)
-        if (topic) {
-          tabs.push(topic)
-        }
-      })
-    }
-    return tabs
-  }, [coreTopics])
-  
+  const frontpageTab = {_id: '0', name: FRONTPAGE_TAB_NAME}
   const [activeTab, setActiveTab] = useState<TopicsBarTab>(frontpageTab)
-  const [leftArrowVisible, setLeftArrowVisible] = useState(false)
-  const [rightArrowVisible, setRightArrowVisible] = useState(true)
-  const navigate = useNavigate();
-  const { location, query } = useLocation()
-  const { captureEvent } = useTracking()
-  const activeCoreTopic = useMemo(
-    () => coreTopics?.find(t => t._id === activeTab._id),
-    [coreTopics, activeTab]
-  );
+  const activeCoreTopic = activeTab._id === frontpageTab._id ? null : activeTab as TagDetailsFragment
```

Comment:
Please add a code comment when using `as` to explain why you need it (or refactor to remove the usage if possible)

Reply:
will do, this is an artifact of us not having the tag collection here anymore. plausibly it's cleaner to just fetch this using `useSingle`

Reply:
Updated to do that, though not quite sure how to test spotlight stuff

Reply:
Oh sorry! Yeah I didn't quite notice the spotlight stuff before - I need to check with our content team to see if we still need this (we've never used it). I would prefer not having an extra `useSingle` every time you change tabs if we can avoid it, so either we remove that bit of code or we update the topic bar component to return the needed field (`introSequenceId`) and then we can remove the `useSingle`

They should get back to me tomorrow. Also if you prefer, you can just hand this PR off to me and I can wrap it up when they reply. At this point the remaining updates are EAF-specific things so I'm happy to take responsibility for them :)

Reply:
> I would prefer not having an extra useSingle

My hope is that this mostly hits the cache bc we've already loaded the relevant things, but I'm not sure how caching actually works though.

Another option is to make the `onTagSelectionUpdated`/ `activeTab` be `TopicsBarTab | TagFragment` and get back to casting (or even let TS derive the type). Casting is valid here bc all those objects are actually tab objects, but relying on that ties us to the internal implementation of the HomeTagBar.

getting rid of spotlight also works if that's an option ofc 😛.

> They should get back to me tomorrow. Also if you prefer, you can just hand this PR off to me and I can wrap it up when they reply. At this point the remaining updates are EAF-specific things so I'm happy to take responsibility for them :)

yeah, feel free to update this accordingly. Thanks for your help getting this in better shape 🙂